app/main.py

```diff
@@ -1,34 +1,99 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row: int = row
+        self.column: int = column
+        self.is_alive: bool = is_alive
+
+    def hit(self) -> None:
+        self.is_alive = False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple[int, int], end: tuple[int, int], is_drowned=False) -> None:
+        self.decks = self._create_decks(start, end)
+        self.is_drowned = is_drowned
```

Comment:
The `_validate_ship` method is defined but never called. You should call this method in the `Ship` constructor to ensure that the ship is placed correctly according to the game rules.